app/assets/javascripts/discourse/tests/acceptance/sidebar-user-community-section-test.js

```diff
@@ -1308,10 +1404,39 @@ acceptance(
 
       assert.true(
         query(
-          ".sidebar-section[data-section-name='community'] .sidebar-section-link[data-link-name='everything']"
+          ".sidebar-section-link[data-link-name='everything']"
         ).href.endsWith("/new"),
         "links to /new because there is 1 new topic"
       );
     });
+
+    test("everything link's href is always the latest topics list when sidebar_link_to_filtered_list is false", async function (assert) {
+      updateCurrentUser({
+        user_option: {
+          sidebar_link_to_filtered_list: false,
+        },
+      });
+      this.container.lookup("service:topic-tracking-state").loadStates([
+        {
+          topic_id: 1,
+          highest_post_number: 1,
+          last_read_post_number: null,
+          created_at: "2022-05-11T03:09:31.959Z",
+          category_id: 1,
+          notification_level: null,
+          created_in_new_period: true,
+          treat_as_new_topic_start_date: "2022-05-09T03:17:34.286Z",
+        },
+      ]);
+
+      await visit("/");
+
+      assert.true(
+        query(
+          ".sidebar-section-link[data-link-name='everything']"
+        ).href.endsWith("/latest"),
+        "everything link href is /latest"
```

Comment:
Just wondering qunit-dom has functions that allows us to assert for the href since you've starrted introducing qunit-dom in this PR.

Reply:
There's indeed a better way to assert for href, see https://github.com/discourse/discourse/pull/22203/commits/e96a3b79f26dd92be4384b161b23de2d080fa2f9.